test/test_decomp.py

```diff
@@ -515,6 +515,62 @@ def check_decomposed(aten_name):
                     "only backwards is decomposed, but dtype doesn't support AD"
                 )
 
+    def test_slice(self):
+        class SampleInput:
+            def __init__(self, input, args, kwargs={}):
+                self.input = input
+                self.args = args
+                self.kwargs = kwargs
+
+        class SliceOpInfo:
+            name = "fake slice"
+            supports_autograd = False
+            decomp_aten_name = "aten.slice.Tensor"
+            def op(self, *args, **kwargs):
+                return torch.ops.aten.slice(*args, **kwargs)
+
+            def get_op(self):
+                return torch.ops.aten.slice
+
+            def sample_inputs(self, device, dtype, requires_grad):
+                for i in [
+                    SampleInput(
+                        input=torch.ones(3, device=device, dtype=dtype, requires_grad=requires_grad),
+                        args=(0,),
+                    ),
+                    SampleInput(
+                        input=torch.randn((20, 30, 40), device=device, dtype=dtype, requires_grad=requires_grad),
+                        args=(),
+                        kwargs={
+                            'dim': 1,
+                            'start': 1,
+                            'end': -2,
+                        }
+                    ),
+                    SampleInput(
+                        input=torch.randn((20, 30, 40), device=device, dtype=dtype, requires_grad=requires_grad),
+                        args=(),
+                        kwargs={
+                            'dim': 1,
+                            'start': 1,
+                            'end': -2,
+                            'step': 3,
+                        }
+                    ),
+                    SampleInput(
+                        input=torch.randn((20, 30, 40), device=device, dtype=dtype, requires_grad=requires_grad),
+                        args=(),
+                        kwargs={
+                            'dim': 0,
+                            'start': -10,
+                            'end': -2,
+                            'step': 2,
+                        }
+                    ),
+                ]:
+                    yield i
+        slice = SliceOpInfo()
+        self.do_cross_ref('cpu', torch.float, slice, run_all=True)
```

Comment:
Weren't you going to move this into common_methods_invocations

Reply:
it blows up spectacularly in https://github.com/pytorch/pytorch/pull/85314. I'd suggest we do it as a follow up if you don't mind. I'm usually pretty good at following up

Reply:
but... like... that the OpInfo isn't working with the other tests implies that the OpInfo is wrong...

Reply:
> but... like... that the OpInfo isn't working with the other tests implies that the OpInfo is wrong...

It seems it just needs way more integration work to run all the tests we run with OpInfo
* there are issues using torch.ops.aten.slice as `OpInfo.op`
* it actually passes for more types than I specified
* it also passes more forward/backward tests than I specified it should be passing.

For testing the forward of `slice` which I think was the main intent of Will's tests, it works pretty well as the part of `decomp` tests?

Reply:
I don't think it is going to be that hard to fix. I'm checking out your PR to try it.